src/libraries/System.Private.CoreLib/src/Microsoft/Win32/SafeHandles/SafeFileHandle.ThreadPoolValueTaskSource.cs

```diff
@@ -116,8 +118,16 @@ private void ExecuteInternal()
                 }
                 finally
                 {
+                    if (_strategy is not null && exception is not null)
+                    {
+                        Debug.Assert(_operation == Operation.Write);
```

Comment:
we do it only for writes as `OSFileStreamStrategy.ReadAsync` is not using this type:

https://github.com/dotnet/runtime/blob/79c4144f174754bf4fa18827a292549214f86581/src/libraries/System.Private.CoreLib/src/System/IO/Strategies/OSFileStreamStrategy.cs#L310-L322

Reply:
> we do it only for writes as OSFileStreamStrategy.ReadAsync is not using this type

Per my previous comment... should it?